### ISSUE TITLE
Recreate MIG VMs after TLS cert update

### ISSUE THREAD

**Issue by paulvoosten** [terraform-google-modules/terraform-google-vault#170]
## TL;DR
As far as I can tell getting a newly created self managed TLS certificate on to the MIG VMs requires a manual action: recreating the VMs.

My question is: is there a way to automate this step?

## Setup
We currently have a GitHub Action that applies a Terraform configuration containing our Vault cluster on GCE and the associated TLS certificate twice a month in order to keep said certificate up to date.

This works fine, except for the above mentioned issue: the newly issued TLS certificate isn't available on the MIG VMs until they have been manually recreated.

### Terraform configuration
Here is a redacted version of the used Terraform configuration.

```hcl
locals {
  vault_domain = "vault.[redacted]"
}

resource "tls_private_key" "vault" {
  algorithm = "RSA"
}

resource "acme_registration" "vault" {
  account_key_pem = tls_private_key.vault.private_key_pem
  email_address   = var.acme_registration_email
}

resource "acme_certificate" "vault" {
  account_key_pem = acme_registration.vault.account_key_pem
  common_name     = local.vault_domain
  dns_challenge {
    provider = "gcloud"
    config = {
      GCE_PROJECT = module.dns_hub_project.project_id
    }
  }
}

resource "google_kms_key_ring" "vault_tls" {
  project  = [redacted]
  name     = "[redacted]"
  location = local.default_region

  depends_on = [
    module.secrets_project
  ]
}

resource "google_kms_crypto_key" "vault_tls" {
  name            = "[redacted]"
  key_ring        = google_kms_key_ring.vault_tls.id
  rotation_period = "604800s"
  version_template {
    algorithm        = "GOOGLE_SYMMETRIC_ENCRYPTION"
    protection_level = "SOFTWARE"
  }
}

resource "google_kms_secret_ciphertext" "vault_tls" {
  crypto_key = google_kms_crypto_key.vault_tls.id
  plaintext  = acme_certificate.vault.private_key_pem
}

resource "google_storage_bucket" "vault_tls" {
  project  = [redacted]
  name     = "[redacted]"
  location = local.default_region
}

resource "google_storage_bucket_object" "vault_srv_crt" {
  name    = "vault.crt"
  bucket  = google_storage_bucket.vault_tls.name
  content = acme_certificate.vault.certificate_pem
}

resource "google_storage_bucket_object" "vault_ca_crt" {
  name    = "ca.crt"
  bucket  = google_storage_bucket.vault_tls.name
  content = acme_certificate.vault.issuer_pem
}

resource "google_storage_bucket_object" "vault_key" {
  name    = "vault.key.enc"
  bucket  = google_storage_bucket.vault_tls.name
  content = google_kms_secret_ciphertext.vault_tls.ciphertext
  # Ciphertext changes on each invocation, so ignore changes
  lifecycle {
    ignore_changes = [
      content,
    ]
  }
}

module "vault" {
  source  = "terraform-google-modules/vault/google"
  version = ">= 6.1.1"

  project_id              = [redacted]
  region                  = local.default_region
  storage_bucket_location = local.default_region
  storage_bucket_class    = "STANDARD"
  kms_keyring             = "[redacted]"
  kms_crypto_key          = "[redacted]"
  domain                  = local.vault_domain
  vault_version           = var.vault_version

  manage_tls              = false
  vault_tls_bucket        = google_storage_bucket.vault_tls.name
  vault_tls_kms_key       = google_kms_crypto_key.vault_tls.id
  vault_ca_cert_filename  = google_storage_bucket_object.vault_ca_crt.output_name
  vault_tls_cert_filename = google_storage_bucket_object.vault_srv_crt.output_name
  vault_tls_key_filename  = google_storage_bucket_object.vault_key.output_name

  service_account_project_additional_iam_roles = ["roles/iam.serviceAccountKeyAdmin"]
}
```
---
